cloud/amazon/redshift_security_group.py

```diff
@@ -0,0 +1,229 @@
+            redshift.revoke_cluster_security_group_ingress(group['ClusterSecurityGroupName'], **group_kwargs)
+
+        # Revoke EC2 security groups.
+        for rule in group['EC2SecurityGroups']:
+            group_kwargs = {
+                'ec2_security_group_name': rule['EC2SecurityGroupName'],
+                'ec2_security_group_owner_id': rule['EC2SecurityGroupOwnerId']
+            }
+            redshift.revoke_cluster_security_group_ingress(group['ClusterSecurityGroupName'], **group_kwargs)
+
+    except boto.exception.JSONResponseError, e:
+        # FIXME: Change this to just the error message when
+        # https://github.com/boto/boto/issues/2776 is fixed.
+        module.fail_json(msg=str(e))
+
+
+def delete_group(module, redshift):
+
+    group_name = module.params.get('group_name')
+
+    try:
+        redshift.delete_cluster_security_group(group_name)
+    except boto.exception.JSONResponseError, e:
+        # FIXME: Change this to just the error message when
+        # https://github.com/boto/boto/issues/2776 is fixed.
+        module.fail_json(msg=str(e))
+
+    return True
+
+
+def create_group(module, redshift):
+
+    group_name = module.params.get('group_name')
+    description = module.params.get('description')
+    rules = module.params.get('rules')
+
+    try:
+        group = []
+
+        sec_groups = redshift.describe_cluster_security_groups(group_name)
+        for sec_group in sec_groups['DescribeClusterSecurityGroupsResponse']['DescribeClusterSecurityGroupsResult']['ClusterSecurityGroups']:
+            if sec_group['ClusterSecurityGroupName'] == group_name:
+                group = sec_group
+                break
+
+        _clear_rules(module, redshift, group)
+    except boto.redshift.exceptions.ClusterSecurityGroupNotFound:
+
+        try:
+            redshift.create_cluster_security_group(group_name, description)
+        except boto.exception.JSONResponseError, e:
+            # FIXME: Change this to just the error message when
+            # https://github.com/boto/boto/issues/2776 is fixed.
+            module.fail_json(msg=str(e))
+
+    # Process all rule properties.
+    if rules is not None:
+        if not isinstance(rules, list):
+            module.fail_json(msg='Rules needs to be a list of CIDR or EC2 security groups')
+
+    try:
+        for rule in rules:
+            group_kwargs = {}
+            if 'cidr' in rule:
+                group_kwargs['cidrip'] = rule['cidr']
+            elif 'sg_name' in rule:
+                group_kwargs['ec2_security_group_name'] = rule['sg_name']
+                if 'sg_owner' in rule:
+                    group_kwargs['ec2_security_group_owner_id'] = rule['sg_owner']
+
+            redshift.authorize_cluster_security_group_ingress(group_name, **group_kwargs)
+
+    except boto.exception.JSONResponseError, e:
+        # FIXME: Change this to just the error message when
+        # https://github.com/boto/boto/issues/2776 is fixed.
```

Comment:
I'm going to suggest that this is never going to happen (all the development effort has gone to boto3). Might as well just delete the FIXME comments across the board.